packages/components/src/components/post-accordion-item/post-accordion-item.tsx

```diff
@@ -55,15 +56,17 @@ export class PostAccordionItem {
     this.validateHeadingLevel();
   }
 
-  // capture to make sure the "collapsed" property is updated before the event is consumed
+  // Capture to make sure the "collapsed" property is updated before the event is consumed
   @Listen('postToggle', { capture: true })
   onCollapseToggle(event: CustomEvent<boolean>): void {
-    if (
-      event.target === this.host &&
-      (event.target as HTMLElement).localName === 'post-accordion-item'
-    ) {
-      this.collapsed = !event.detail;
-    }
+    eventGuard.call(
+      this,
+      event,
+      { targetLocalName: 'post-accordion-item', delegatorSelector: 'post-accordion-item' },
```

Comment:
Are you sure, we need to check the delegatorSelector here? And if so, does the selector need to be 'post-accordion-item'?
This would mean, the event must be triggered from a `post-accordion-item` and its closest parent which listens to the event, must be a `post-accordion-item` as well.